src/main/java/com/minecolonies/coremod/entity/ai/mobs/AbstractArcherBarbarian.java

```diff
@@ -0,0 +1,284 @@
+package com.minecolonies.coremod.entity.ai.mobs;
+
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.entity.EntityCitizen;
+import net.minecraft.entity.*;
+import net.minecraft.entity.ai.*;
+import net.minecraft.entity.monster.EntityMob;
+import net.minecraft.entity.passive.EntityWolf;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.projectile.EntityArrow;
+import net.minecraft.entity.projectile.EntityTippedArrow;
+import net.minecraft.init.Blocks;
+import net.minecraft.init.Items;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.datasync.DataParameter;
+import net.minecraft.network.datasync.DataSerializers;
+import net.minecraft.network.datasync.EntityDataManager;
+import net.minecraft.stats.AchievementList;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.DifficultyInstance;
+import net.minecraft.world.EnumDifficulty;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import javax.annotation.Nullable;
+import java.util.Calendar;
+
+/**
+ * Abstract Class implementing the Archer Barbarian
+ */
+public abstract class AbstractArcherBarbarian extends EntityMob implements IRangedAttackMob
+{
+    /* default */ final private Colony colony = ColonyManager.getClosestColony(world, this.getPosition());
+    private static final DataParameter<Boolean>           SWINGING_ARMS     = EntityDataManager.<Boolean>createKey(AbstractArcherBarbarian.class, DataSerializers.BOOLEAN);
+    private final        EntityAIAttackRangedBowBarbarian aiArrowAttack     = new EntityAIAttackRangedBowBarbarian(this, 1.0D, 20, 15.0F);
+    private final        EntityAIAttackMelee              aiAttackOnCollide = new EntityAIAttackMelee(this, 1.2D, false)
+    {
+        /**
+         * Resets the task
+         */
+        public void resetTask()
+        {
+            super.resetTask();
+            AbstractArcherBarbarian.this.setSwingingArms(false);
+        }
+
+        /**
+         * Execute a one shot task or start executing a continuous task
+         */
+        public void startExecuting()
+        {
+            super.startExecuting();
+            AbstractArcherBarbarian.this.setSwingingArms(true);
+        }
+    };
+
+    public AbstractArcherBarbarian(final World worldIn)
+    {
+        super(worldIn);
+        this.setSize(0.6F, 1.99F);
+        this.setCombatTask();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this call from a constructor to the overridable "setCombatTask" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)